gui/ros_gui/gui/configpanel/configpanel.py

```diff
@@ -0,0 +1,357 @@
+                case ParameterType.PARAMETER_STRING:
+                    self.parameters[param_name] = future.result().values[idx].string_value
+                case ParameterType.PARAMETER_BYTE_ARRAY:
+                    self.parameters[param_name] = future.result().values[idx].byte_array_value
+                case ParameterType.PARAMETER_BOOL_ARRAY:
+                    self.parameters[param_name] = future.result().values[idx].bool_array_value
+                case ParameterType.PARAMETER_INTEGER_ARRAY:
+                    self.parameters[param_name] = future.result().values[idx].integer_array_value
+                case ParameterType.PARAMETER_DOUBLE_ARRAY:
+                    self.parameters[param_name] = future.result().values[idx].double_array_value
+                case ParameterType.PARAMETER_STRING_ARRAY:
+                    self.parameters[param_name] = future.result().values[idx].string_array_value
+                case _:
+                    self.get_logger().info(f'Parameter {param_name} has an unsupported type {param_type}')
+
+    async def pick_file(self, node_name, param_name) -> None:
+        """ Opens a file picker and sets the selected file as the parameter value.
+        
+        Args:
+            node_name (str): Name of the node that the parameter belongs to.
+            param_name (str): Name of the parameter.
+        """
+        result = await local_file_picker('~/.astazero/ATOS', multiple=False, show_hidden_files=True)
+        ui.notify(f'You selected {result}')
+        if not result:
+            return
+        self.parameters[param_name] = str(result)
+        self.set_parameter(node_name, param_name, str(result))
+
+    async def pick_files(self, node_name, param_name) -> None:
+        """ Opens a file picker and sets the selected files as the parameter value.
+
+        Args:
+            node_name (str): Name of the node that the parameter belongs to.
+            param_name (str): Name of the parameter.
+        """
+        result = await local_file_picker('~/.astazero/ATOS', multiple=True, show_hidden_files=True)
+        ui.notify(f'You selected {result}')
+        if not result:
+            return
+        self.parameters[param_name] = str(result)
+        self.set_parameter(node_name, param_name, str(result))
+
+    def set_param_callback(self, future, param_name) -> None:
+        """ Callback function for the set parameters service call. Notifies the user if the call was successful or not.
+        
+        Args:
+            future (Future): Future object returned by the set parameters service call (Bool).
+            param_name (str): Name of the parameter.
+        """
+        try:
+            response = future.result()
+        except Exception as e:
+            self.get_logger().info('Service call failed %r' % (e,))
+            return
+        self.get_logger().debug(f'Setting parameter {param_name} was {"successful" if response.result.successful else "unsuccesful"}')
```

Comment:
I think it is enough to let the user be notified on error. The successful case should be presumed.

Reply:
I disagree. The other notification was superfluous, but this one is needed imo. From a user perspective, it's nice to get any form of validation that an action has been made (in this case you also get information the action was successful). If the user wouldn't get this response, he/she wouldn't notice the difference between pressing "enter" to submit or just deselecting the input field. For a new user, I also think that it's a bit too much to explicitly write out that you should submit by pressing enter, it's better for them to learn it by performing the action and seeing a response vs no response for other actions (like deselecting)